### PR TITLE
Handle missing process.env correctly

### PR DESCRIPTION
**Pull request by ptpaterson**
Ticket(s): FE-3607

## Problem
We use `process.env` to fallback on env vars for client secret, but we are blowing up in environments where `process.env` is undefined.

A customer using Vue & Nuxt encountered an issue where creating a `Client` instance breaks Vue reactivity. The issue is triggered because `process` is defined to be something like this:

```javascript
{ dev: true, test: false, server: false, client: true }
```

## Solution
Check for `process.env` is what we actually want

## Result
We can use the driver with Nuxt.js

## Out of scope
Adding Nuxt.js platform testing.

## Testing
Local testing with a minimal Vue.js app. Create the simplest app:

```shell
npx nuxi@latest init my-app
```

Update the `app.vue` file:

```vue
<template>
  <div>
    <input type="text" v-model="title" />

    {{ title }}
    {{ data }}
  </div>
</template>

<script setup>

const title = ref('');
const data = ref('')

import { Client, fql } from "fauna";

// creating the client will break the app before this fix
const client = new Client({
  secret: YOUR_FAUNA_KEY,
});

client.query(fql`"hello world"`).then(res => data.value = res.data)
</script>
```

start the app and see the request complete and reactivity when you type into the input field.

```shell
npm run dev
```

if linking installing a local version of the driver, Nuxt doesn't like symlinks, so add this to nuxt.config.ts

```javascript
  vite: {
    server: {
      fs: {
        allow: ['/Users/psquared/Documents/fauna/fauna-js'],
      },
    },
  },
  imports: {
    transform: {
      exclude: [/\bfauna\b/]
    }
  },
```

----
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/client.ts [hunk 1]
```diff
@@ -296,10 +296,11 @@ export class Client {
...
 
   #getSecret(partialClientConfig?: ClientConfiguration): string {
-    let fallback = undefined;
-    if (typeof process === "object") {
-      fallback = process.env["FAUNA_SECRET"];
-    }
+    const fallback =
+      process?.env !== null && typeof process.env === "object"
+        ? process.env["FAUNA_SECRET"]
+        : undefined;
```
Comment (cleve-fauna) [inline src/client.ts:302]: can we just do this?

```suggestion
    const fallback = process?.env?.FAUNA_SECRET;
```

File: src/util/environment.ts [hunk 2]
```diff
@@ -240,7 +240,7 @@ const crossGlobal =
  */
 const getNodeRuntimeEnv = (): string => {
   // return early if process variables are not available
-  if (!(typeof process !== "undefined" && process && process.env)) {
+  if (!(process?.env !== null && typeof process.env === "object")) {
```
Comment (cleve-fauna) [inline src/util/environment.ts:243]:
```suggestion
  if (typeof process?.env !== "object") {
```

**Review by cleve-fauna — COMMENTED** [review]
some simplifications

**Review by cleve-fauna — COMMENTED** [review]
some simplifications

File: src/util/environment.ts [hunk 3]
```diff
@@ -240,7 +240,7 @@ const crossGlobal =
  */
 const getNodeRuntimeEnv = (): string => {
   // return early if process variables are not available
-  if (!(typeof process !== "undefined" && process && process.env)) {
+  if (typeof process?.env !== "object") {
```
Comment (cleve-fauna) [inline src/util/environment.ts:243]:
```suggestion
  if (typeof process?.env !== "object" || process.env === null) {
```

**Comment by ptpaterson** [comment]
yeeeeea that was much simpler.

File: src/util/environment.ts (hunk 3 above)
Comment (cleve-fauna) [reply src/util/environment.ts:243]: null's type is object

**Review by cleve-fauna — APPROVED** [review]
Ship if you've integ tested using the process you described in the description.

**Pull request merged**